### PR TITLE
Add support for Operation, OperationParameter and SchemaField filtering

### PR DESCRIPTION
**Pull request by aromanet42**
Fix for https://github.com/jrcodeza/spring-openapi/issues/41

### CODE REVIEW

File: schema-generator-plugin/src/main/java/com/github/jrcodeza/schema/generator/plugin/GenerateOpenApiSchemaMojo.java [hunk 1]
```diff
@@ -104,6 +119,14 @@ public void execute() {
 		}
 	}
 
+	private <T> T parseInputFilter(String className, Class<T> clazz) {
+		if (className == null || className.isEmpty()) {
```
Comment (jrcodeza) [inline schema-generator-plugin/src/main/java/com/github/jrcodeza/schema/generator/plugin/GenerateOpenApiSchemaMojo.java]: You can use StringUtils.isNotEmpty(className) it does the null check and empty check automatically

File: schema-generator/src/main/java/com/github/jrcodeza/schema/generator/OpenAPIGenerator.java [hunk 2]
```diff
@@ -63,27 +68,39 @@
...
     }
 
     public OpenAPIGenerator(List<String> modelPackages, List<String> controllerBasePackages, Info info,
                             List<SchemaInterceptor> schemaInterceptors,
                             List<SchemaFieldInterceptor> schemaFieldInterceptors,
                             List<OperationParameterInterceptor> operationParameterInterceptors,
                             List<OperationInterceptor> operationInterceptors,
-                            List<RequestBodyInterceptor> requestBodyInterceptors) {
+                            List<RequestBodyInterceptor> requestBodyInterceptors,
+                            OperationFilter operationFilter,
```
Comment (jrcodeza) [inline schema-generator/src/main/java/com/github/jrcodeza/schema/generator/OpenAPIGenerator.java]: I would add another constructor. So this one would remain as is and new one would have thee 3 filters. Then the smaller one can call the complete one where these 3 additional filters would be null. It's for sake of backwards compatibility

File: schema-generator-plugin/src/main/java/com/github/jrcodeza/schema/generator/plugin/GenerateOpenApiSchemaMojo.java (hunk 1 above)
Comment (aromanet42) [reply schema-generator-plugin/src/main/java/com/github/jrcodeza/schema/generator/plugin/GenerateOpenApiSchemaMojo.java]: ok I changed that

File: schema-generator/src/main/java/com/github/jrcodeza/schema/generator/OpenAPIGenerator.java (hunk 2 above)
Comment (aromanet42) [reply schema-generator/src/main/java/com/github/jrcodeza/schema/generator/OpenAPIGenerator.java]: yes you're right, I didn't think about backwards compatibility. I added a commit to fix that.

**Pull request merged**